redbot/core/utils/chat_formatting.py

```diff
@@ -429,3 +430,44 @@ def humanize_timedelta(
             strings.append(f"{period_value} {unit}")
 
     return ", ".join(strings)
+
+
+def _convertlocale(locale: str) -> str:
+    """
+    Convert a locale string to lang_territory format
+
+    Parameters
+    ----------
+    locale : str
+        Locale to be converted into a babel supported format.
+
+    Returns
+    -------
+    str
+        Babel compatible locale.
+    """
+    formatting = ""
+    if "." in locale:
+        locale, formatting = locale.split(".")
+    if "-" in locale:
+        locale = locale.replace("-", "_", 1)
+    if formatting:
+        locale = ".".join([locale, formatting])
+    return locale
+
+
+def humanize_int(val: Union[int, float]) -> str:
+    """
+    Convert an int to a str with digit separators based on bot locale
+
+    Parameters
+    ----------
+    val : Union[int, float]
+        The int to be formatted.
+
+    Returns
+    -------
+    str
+        locale aware formated number.
+    """
+    return format_decimal(val, locale=_convertlocale(get_locale()))
```

Comment:
There's a chance the locale being used isn't actually known to Babel, in which case it will throw a `babel.core.UnknownLocaleError` - when this happens, simply default to `en_US` :smiley:

Reply:
Just for wider visibility,

I changed the code up a little bit.

But basically it uses the parses provided by babel (no need to reinvent the wheel) 
It first tries to parse the red locale, if it fails with `ValueError`, then it tries to get the closest match, if  it  fails (`ValueError` or `TypeError` or `babel.core.UnknownLocaleError`) or can't get a close match then it defaults to  `en_US`

Red's locale and Babel supported locale is then cached and only updated when Red's locale is changed
This avoid calling the babel parser and Try/Except blocks every single time we call `humanize_int`

Tried with about 100 different locale formats and not once it was defaulted to `en_US` :D 

Note to self: Stop trying to reinvent the wheel